pysat/utils/time.py

```diff
@@ -73,45 +74,77 @@ def parse_date(str_yr, str_mo, str_day, str_hr='0', str_min='0', str_sec='0',
     return out_date
 
 
-def calc_freq(index):
-    """ Determine the frequency for a time index
+def calc_res(index, use_mean=False):
+    """ Determine the resolution for a time index
 
     Parameters
     ----------
     index : array-like
         Datetime list, array, or Index
+    use_mean : bool
```

Comment:
Would it be practical to support mode and median as well? Perhaps allow users to pass in their own favorite function or something.

Since the last review I've been a little uncomfortable with only a single averaging function since instrument outputs can all have all kinds of time jumps. We won't get all of them ahead of time but a user should have better knowledge of the specific data issues they face.

Reply:
It's never practical to support the mode.  It wouldn't be hard to allow users to pass their own function in, but the question for this might be: "do we need this functionality?".  This should be raised as an issue, because it's well beyond the scope here.